src/journal/journald-server.c

```diff
@@ -1175,7 +1180,10 @@ void server_dispatch_message(
                                       NULL);
 
 finish:
-        dispatch_message_real(s, iovec, n, m, ucred, tv, label, label_len, unit_id, priority, object_pid);
+        /* restore cgroup path for logging */
+        if (c)
+                *c = '/';
+        dispatch_message_real(s, iovec, n, m, ucred, tv, label, label_len, unit_id, priority, object_pid, path);
```

Comment:
Is it really safe to pass `path` here if `c == NULL`? In that case it was not a ≥ 3 component path and `path` contains something undefined. Maybe pass `c ? path : NULL`?

I don't know the details of this though, so this could well work fine.